api/v1beta1/disruption_webhook.go

```diff
@@ -138,15 +142,45 @@ func (r *Disruption) ValidateCreate() error {
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *Disruption) ValidateUpdate(old runtime.Object) error {
-	logger.Debugw("validating updated disruption", "instance", r.Name, "namespace", r.Namespace)
+	logger := logger.With("instance", r.Name, "namespace", r.Namespace)
+	logger.Debugw("validating updated disruption")
+
+	var err error
+
+	oldDisruption := old.(*Disruption)
+
+	// ensure finalizer removal is only allowed if no related chaos pods exists
+	// such verification is only needed if disruption is currently in removal as controller will always add finalizer to disruptions not in removal during the reconcile loop
+	// we should NOT always prevent finalizer removal because chaos controller reconcile loop will go through this mutating webhook when perfoming updates
+	// and need to be able to remove the finalizer to enable the disruption to be garbage collected on successful removal
+	if !oldDisruption.ObjectMeta.DeletionTimestamp.IsZero() && controllerutil.ContainsFinalizer(oldDisruption, chaostypes.DisruptionFinalizer) && !controllerutil.ContainsFinalizer(r, chaostypes.DisruptionFinalizer) {
```

Comment:
Why use `oldDisruption.ObjectMeta.DeletionTimestamp` here instead of `oldDisruption.DeletionTimestamp` given similar changes elsewhere?